cpp/src/arrow/extension/json.h

```diff
@@ -27,6 +27,11 @@
 
 namespace arrow::extension {
 
+class ARROW_EXPORT JsonArray : public ExtensionArray {
+ public:
+  using ExtensionArray::ExtensionArray;
+};
```

Comment:
This is not used in C++, but I think also not actually needed for the Python bindings?

Reply:
Indeed it is not. Removed.